### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by MaximGorbach**
Added some documentation for the engine for a user.

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+    if (engine.KeyStates.isPressed("Space")) {
+      this.Velocity.y = this.jump_speed;
+      this.times_jumped++;
+    }
+  }
+}
+```
+
+### Component Scripts
+Component scripts define custom components which can be added to entities, either programmatically or through the editor. Component scripts must extend from the [`ComponentBase`](docs/classes/ComponentBase.md) class. The `ComponentBase` class provides a number of useful functions which can be overridden:
```
Comment (Louis-Tarvin) [inline Readme.md:59]: The base class is just called `Component`, not `ComponentBase`. This link gives a 404

File: Readme.md [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+* `onCreate`: Called when the owner of the component is added to a scene.
+* `onDelete`: Called when this component is deleted.
```
Comment (Louis-Tarvin) [inline Readme.md:61]:
```suggestion
* `onDelete`: Called when this component is removed or the owner is deleted.
```

File: Readme.md [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+    return Math.sqrt((this.x - other.x) ** 2 + (this.y - other.y) ** 2);
+  }
+}
+```
+
+### System Scripts
+System scripts define custom systems which can be added to scenes which act on a component. Systems are added to a scene automatically if an entity with the desired component is added to the scene. They can also be added manually through the [`Scene`](docs\classes\Scene.md) or [`SceneManager`](docs\classes\SceneManager.md) class.
+> Note that currently the system name must match the component name to be added automatically.
+> 
+System scripts must extend from the [`SystemBase`](docs/classes/SystemBase.md) class. The `SystemBase` class provides the `update(entities)` function which is called every frame. The `entities` parameter is an array of entities which have the component that the system is associated with. To define the component that the system is associated with, you must set the `component` property of the class to the component class.
```
Comment (Louis-Tarvin) [inline Readme.md:87]: Again, this link 404s because the base class is just called `System` not `SystemBase`

File: Readme.md [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+Pixelot supports rendering tilemap layers. Tilemaps should be created in 'Tiled' (https://www.mapeditor.org/). In order to export in the correct format, ensure that the tilesets are embedded in the `json` file (can be toggled when opening a tileset in Tiled). Additionally, make sure to export it directly into the projects directory, as Tiled will configure the paths to the tilesets relative to the `json` file. As such, also make sure that the tileset images are being stored within the projects directory. **This means that if you want to move the location of the tilemap `.json`, or the tileset images, you should delete it and re-export from Tiled to ensure the relative paths stay valid**
+
+Pixelot supports arbitrary layers and tilesets, but won't support any custom properties, or transparency colours set in Tiled. In order to add transparency, make tilesets with transparent backgrounds in an image editor, and then export them as `.png` files.
+
+To retrieve objects from the object layer, use the `getObject` function of the `TileMapObjectLayer` class. 
+To retrieve an object layer from the tilemap, use the `getObjectLayer` function of the `TileMapLayer` class.
+
+
+### Shaders
```
Comment (Louis-Tarvin) [inline Readme.md:180]: Empty subheading?

**Review by Louis-Tarvin — APPROVED** [review]
Very nice. We should try to keep this up to date from now on.
Spotted a couple of minor errors.

File: Readme.md (hunk 4 above)
Comment (MaximGorbach) [reply Readme.md:180]: Yeah I left this in for when we eventually would add the ability to apply shaders through the UI.

**Pull request merged**